public/sketch.js

```diff
@@ -1,116 +1,82 @@
+
+  for (const [index, option] of Object.entries(poll.options)) {
     let count = poll.votes[index];
     let width = map(count, 0, maxVotes, 0, 100);
-    if (firstRender) {
-      //Container div per option
-      let optionDiv = document.createElement('div');
-      optionDiv.classList.add('option');
-      // Add option text on a separate line, so it can get any length
-      let optionText = document.createElement('span');
-      optionText.innerHTML = option;
-      optionDiv.append(optionText);
-      resultsDiv.append(optionDiv);
-
-      //Add option progress bar
-      let progressBar = document.createElement('div');
-      progressBar.id = 'progressBar_' + index;
-      progressBar.classList.add('progressBar');
-      progressBar.style.setProperty('width', width + '%');
-      progressBar.innerHTML = Math.round(count/totalVotes*100) + '%';
-      resultsDiv.append(progressBar);
-    } else {
-      let progressBar = document.getElementById("progressBar_" + index);
-      progressBar.style.setProperty('width', width + '%');
-      progressBar.innerHTML = Math.round(count/totalVotes*100) + '%';
-    }
-  }
 
+    if (firstRender)
+      doTheFirstRender(index, option, resultsDiv);
+
+    // Get the progress bar element
+    let progressBar = document.getElementById("progressBar_" + index);
+    // Set the width
+    progressBar.style.setProperty('width', width + '%');
+    // Set the text
+    progressBar.innerHTML = Math.round(count / totalVotes * 100) + '%';
+  }
   firstRender = false;
 }
 
-async function setup() {
-  noCanvas();
-  await countVotes();
-  setInterval(countVotes, 500);
+const doTheFirstRender = (index, option, resultsDiv) => {
+  // Container div per option
+  let optionDiv = document.createElement('div');
+  optionDiv.classList.add('option');
 
-  let pollQ = createElement('p', poll.question)
-  pollQ.addClass("question")
+  // Add option text on a separate line, so it can get any length
+  let optionText = document.createElement('span');
 
-  radio = createRadio();
-  for (let i = 0; i < poll.options.length; i++) {
-    radio.option(i, poll.options[i]) // First arg is index, second arg is what is visible to user 
-  }
-  // radio.style('width', '50px'); // Change this for width of radio 
-  voteButton = createButton('vote');
-  voteButton.addClass("VoteBTN")
-  voteButton.mousePressed(submitVote);
-}
+  optionText.innerHTML = option;
+  optionDiv.append(optionText);
+  resultsDiv.append(optionDiv);
 
-async function submitVote() {
-  let choice = radio.value(); // choice is a number
-  if (!isNaN(choice)) {
-    let response = await fetch(`vote/${poll_id}/${choice}`);
-    let status = await response.json();
-    console.log(status);
-  } else {
-    console.log('no choice selected');
-  }
+  // Add option progress bar
+  let progressBar = document.createElement('div');
+
+  progressBar.id = 'progressBar_' + index;
+  progressBar.classList.add('progressBar');
+  resultsDiv.append(progressBar);
```

Comment:
I guess here too! Maybe we should just use native JS everywhere? No preference here!

Reply:
Nah, p5 looks prettier and more readable, and also simpler.